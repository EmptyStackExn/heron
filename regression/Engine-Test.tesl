// Using tag relations for modeling engine ignition
// http://wwwdi.supelec.fr/software/TESL/GalleryEngine

Q-clock realtime    // in seconds
Q-clock crankshaft  // in degrees
Q-clock camshaft    // in degrees
tag relation crankshaft = 2.0 * camshaft + 0.0 // crankshaft turns twice as fast as camshaft

tag relation crankshaft = 12000.0 * realtime + 0.0

Q-clock top_dead_center periodic 1. offset 0.
tag relation crankshaft = 360.0 * top_dead_center + 0.0

U-clock exhaust_tdc
top_dead_center every 2 implies exhaust_tdc
U-clock compress_tdc
top_dead_center every 2 starting at 1 implies compress_tdc

U-clock ignition
exhaust_tdc time delayed by <11/500> on realtime implies ignition

@policy asap
@maxstep 5
@dumpres
@run
@output tikz select realtime crankshaft camshaft top_dead_center exhaust_tdc compress_tdc ignition