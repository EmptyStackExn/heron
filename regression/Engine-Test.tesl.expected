  \node       at (2+0.1, -3+0.1)  {\Cross} ;
  \node       at (2+0.1, -4+0.1)  {\Cross} ;
  \node       at (3+0.1, -4+0.1)  {\Cross} ;
  \node       at (5+0.1, -3+0.1)  {\Cross} ;
  \node       at (5+0.1, -4+0.1)  {\Cross} ;
  \node[tick] at (1, -3)  {} ;
  \node[date] at (1, -3)  {0.} ;
  \node[date] at (1, -1)  {0.} ;
  \node[date] at (1, -2)  {0.} ;
  \node[date] at (1, 0)  {0.} ;
  \node[tick] at (1, -4)  {} ;
  \node[date] at (2, 0)  {0.022} ;
  \node[tick] at (2, -6)  {} ;
  \node[date] at (2, -1)  {264.} ;
  \node[date] at (2, -2)  {132.} ;
  \node[date] at (2, -3)  {0.733333333333} ;
  \node[date] at (3, -3)  {1.} ;
  \node[tick] at (3, -3)  {} ;
  \node[date] at (3, -1)  {360.} ;
  \node[date] at (3, -2)  {180.} ;
  \node[date] at (3, 0)  {0.03} ;
  \node[tick] at (3, -5)  {} ;
  \node[date] at (4, -3)  {2.} ;
  \node[tick] at (4, -3)  {} ;
  \node[date] at (4, -1)  {720.} ;
  \node[date] at (4, -2)  {360.} ;
  \node[date] at (4, 0)  {0.06} ;
  \node[tick] at (4, -4)  {} ;
  \node[date] at (5, 0)  {0.082} ;
  \node[tick] at (5, -6)  {} ;
  \node[date] at (5, -1)  {984.} ;
  \node[date] at (5, -2)  {492.} ;
  \node[date] at (5, -3)  {2.73333333333} ;
